compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -27,12 +27,20 @@ use crate::ty;
 
 /// Functionality required for the bytes of an `Allocation`.
 pub trait AllocBytes: Clone + fmt::Debug + Deref<Target = [u8]> + DerefMut<Target = [u8]> {
+    /// Used for giving extra metadata on creation. Miri uses this to allow
+    /// machine memory to be allocated separately from other allocations.
+    type ByteMetadata: Clone + fmt::Debug;
+
     /// Create an `AllocBytes` from a slice of `u8`.
-    fn from_bytes<'a>(slice: impl Into<Cow<'a, [u8]>>, _align: Align) -> Self;
+    fn from_bytes<'a>(
+        slice: impl Into<Cow<'a, [u8]>>,
+        _align: Align,
+        _dsc: Self::ByteMetadata,
```

Comment:
Please pick a consistent name: is this metadata or is it a descriptor?

I would have used neither term and called in `AllocParams`.

Reply:
Yeah ok I was all over the place with the names, initially I wanted to call it the discriminant but a similar name was already taken. I'll use yours though ^^